src/internal_modules/roc_sndio/pump.cpp

```diff
@@ -81,6 +83,26 @@ void Pump::stop() {
     stop_ = 1;
 }
 
+status::StatusCode Pump::close_device_(IDevice* device) {
+    if (device) {
+        return device->close();
+    }
+    return status::StatusOK;
+}
+
+status::StatusCode Pump::close_all_devices_() {
+    IDevice* devices[] = { &main_source_, &sink_, backup_source_, current_source_ };
```

Comment:
current_source_ should not be here, because it points to one of the other two sources, and thus we'd close one of them two times.